cmd/restic/find.go

```diff
@@ -11,31 +11,39 @@ import (
 
 // initMultiSnapshotFilter is used for commands that work on multiple snapshots
 // MUST be combined with restic.FindFilteredSnapshots or FindFilteredSnapshots
+// MUST be followed by finalizeSnapshotFilter after flag parsing
 func initMultiSnapshotFilter(flags *pflag.FlagSet, filt *restic.SnapshotFilter, addHostShorthand bool) {
 	hostShorthand := "H"
 	if !addHostShorthand {
 		hostShorthand = ""
 	}
-	flags.StringArrayVarP(&filt.Hosts, "host", hostShorthand, nil, "only consider snapshots for this `host` (can be specified multiple times) (default: $RESTIC_HOST)")
+	flags.StringArrayVarP(&filt.Hosts, "host", hostShorthand, nil, "only consider snapshots for this `host` (can be specified multiple times, use empty string to override $RESTIC_HOST) (default: $RESTIC_HOST)")
 	flags.Var(&filt.Tags, "tag", "only consider snapshots including `tag[,tag,...]` (can be specified multiple times)")
 	flags.StringArrayVar(&filt.Paths, "path", nil, "only consider snapshots including this (absolute) `path` (can be specified multiple times, snapshots must include all specified paths)")
-
-	// set default based on env if set
-	if host := os.Getenv("RESTIC_HOST"); host != "" {
-		filt.Hosts = []string{host}
-	}
 }
 
 // initSingleSnapshotFilter is used for commands that work on a single snapshot
 // MUST be combined with restic.FindFilteredSnapshot
+// MUST be followed by finalizeSnapshotFilter after flag parsing
 func initSingleSnapshotFilter(flags *pflag.FlagSet, filt *restic.SnapshotFilter) {
-	flags.StringArrayVarP(&filt.Hosts, "host", "H", nil, "only consider snapshots for this `host`, when snapshot ID \"latest\" is given (can be specified multiple times) (default: $RESTIC_HOST)")
+	flags.StringArrayVarP(&filt.Hosts, "host", "H", nil, "only consider snapshots for this `host`, when snapshot ID \"latest\" is given (can be specified multiple times, use empty string to override $RESTIC_HOST) (default: $RESTIC_HOST)")
 	flags.Var(&filt.Tags, "tag", "only consider snapshots including `tag[,tag,...]`, when snapshot ID \"latest\" is given (can be specified multiple times)")
 	flags.StringArrayVar(&filt.Paths, "path", nil, "only consider snapshots including this (absolute) `path`, when snapshot ID \"latest\" is given (can be specified multiple times, snapshots must include all specified paths)")
+}
 
-	// set default based on env if set
-	if host := os.Getenv("RESTIC_HOST"); host != "" {
-		filt.Hosts = []string{host}
+// finalizeSnapshotFilter applies RESTIC_HOST default only if --host flag wasn't explicitly set.
+// This allows users to override RESTIC_HOST by providing --host="" or --host with explicit values.
+func finalizeSnapshotFilter(flags *pflag.FlagSet, filt *restic.SnapshotFilter) {
+	// Only apply RESTIC_HOST default if the --host flag wasn't changed by the user
+	if !flags.Changed("host") {
```

Comment:
There's no need to look at `flags` to detect whether `--host` was specified. `filt.Host` is now `nil` by default and has a different value if and only if `--host` was passed. That is, `filt.Hosts == nil` is sufficient here.

Reply:
Completed.